process/elasticproc/logsevents/serializeDelegators.go

```diff
@@ -26,6 +27,36 @@ func (lep *logsAndEventsProcessor) SerializeDelegators(delegators map[string]*da
 	return nil
 }
 
+// PrepareDelegatorsQueryInCaseOfRevert will prepare the delegators query in case of revert
+func (lep *logsAndEventsProcessor) PrepareDelegatorsQueryInCaseOfRevert(timestamp uint64) *bytes.Buffer {
+	codeToExecute := `
+	if ( !ctx._source.containsKey('unDelegateInfo') ) { return } 
+	if ( ctx._source.unDelegateInfo.length == 0 ) { return }
+
+	Iterator itr = ctx._source.unDelegateInfo.iterator();
```

Comment:
not sure the Java version underneath, but if Java 8+, you could do:
```java
ctx._source.unDelegateInfo.removeIf(info -> info.timestamp.equals(params.timestamp))
```

but can remain as it is

Reply:
it is possible. it's up to you if this worths the refactoring:
https://www.elastic.co/guide/en/elasticsearch/painless/8.6/painless-lambdas.html

Reply:
changed